pkg/daemon/daemon.go

```diff
@@ -822,6 +837,64 @@ func tryEnableRdma() (bool, error) {
 	return false, err
 }
 
+func tryCreateSwitchdevUdevRule(nodeState *sriovnetworkv1.SriovNetworkNodeState) error {
+	glog.V(2).Infof("tryCreateSwitchdevUdevRule()")
+	var new_content string
+	filePath := "/host/etc/udev/rules.d/20-switchdev.rules"
+
+	for _, ifaceStatus := range nodeState.Status.Interfaces {
+		if ifaceStatus.EswitchMode == sriovnetworkv1.ESWITCHMODE_SWITCHDEV {
+			switchID, err := utils.GetPhysSwitchID(ifaceStatus.Name)
+			if err != nil {
+				return err
+			}
+			portName, err := utils.GetPhysPortName(ifaceStatus.Name)
+			if err != nil {
+				return err
+			}
+			new_content = new_content + fmt.Sprintf("SUBSYSTEM==\"net\", ACTION==\"add|move\", ATTRS{phys_switch_id}==\"%s\", ATTR{phys_port_name}==\"pf%svf*\", IMPORT{program}=\"/etc/udev/switchdev-vf-link-name.sh $attr{phys_port_name}\", NAME=\"%s_$env{NUMBER}\"\n", switchID, strings.TrimPrefix(portName, "p"), ifaceStatus.Name)
+		}
+	}
+
+	old_content, err := ioutil.ReadFile(filePath)
+	// if old_content = new_content, don't do anything
+	if err == nil && new_content == string(old_content) {
```

Comment:
you can string(old_content) once. currently you doing in L861 and L866

Reply:
I don't see how I can do it once w/o adding a new string var, any idea?